app/src/main/java/vn/mbm/phimp/me/opencamera/Preview/Preview.java

```diff
@@ -1372,7 +1372,27 @@ public void onFaceDetection(CameraController.Face[] faces) {
 			if( MyDebug.LOG )
 				Log.d(TAG, "saved color effect: " + value);
 
-			CameraController.SupportedValues supported_values = camera_controller.setColorEffect(value);
+            final int[] colorNumber = {0};
+            CameraActivity.toggle.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    String[] colorEffect = getResources().getStringArray(R.array.color_effects);
+                    colorNumber[0]++;
+                    if(colorNumber[0] == 15)
+                        colorNumber[0] = 0;
+                    CameraController.SupportedValues supported_values = camera_controller.setColorEffect(colorEffect[colorNumber[0]]);
```

Comment:
You can get a list of supported color by using **preview.getSupportedColorEffects()**  instead of hardcoding the effects because may be some effect will not work on any other phones.